File: backend/src/lib/roles.ts
import type { Role } from '@annex/shared/types'

// Mirrors frontend/src/lib/rbac.ts — the server is the source of truth for data access.
export const ADMIN: Role[] = ['super_admin', 'company_admin', 'hr_officer']
export const EXEC: Role[] = [...ADMIN, 'ceo']
export const LEADERS: Role[] = [...EXEC, 'manager']
export const PAYROLL: Role[] = [...EXEC, 'finance']

export const isAdmin = (r: Role) => ADMIN.includes(r)
export const isExec = (r: Role) => EXEC.includes(r)
export const isLeader = (r: Role) => LEADERS.includes(r)
export const canSeePay = (r: Role) => PAYROLL.includes(r)

File: backend/api/roles.py
"""Role groups — the server is the source of truth for data access (mirrors frontend/src/lib/rbac.ts)."""

from typing import Literal

Role = Literal["super_admin", "company_admin", "hr_officer", "manager", "employee", "consultant", "finance", "ceo"]
ROLES: tuple[str, ...] = ("super_admin", "company_admin", "hr_officer", "manager", "employee", "consultant", "finance", "ceo")

ADMIN = ("super_admin", "company_admin", "hr_officer")
EXEC = (*ADMIN, "ceo")
LEADERS = (*EXEC, "manager")
PAYROLL = (*EXEC, "finance")


def is_admin(role: str) -> bool:
    return role in ADMIN


def is_exec(role: str) -> bool:
    return role in EXEC


def is_leader(role: str) -> bool:
    return role in LEADERS


def can_see_pay(role: str) -> bool:
    return role in PAYROLL
